sda-commons-starter-kafka/src/main/java/org/sdase/commons/spring/boot/kafka/SdaKafkaProducerConfiguration.java

```diff
@@ -27,9 +30,17 @@ public class SdaKafkaProducerConfiguration {
           ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
           MetadataContextProducerInterceptor.class.getName());
 
+  private final ObjectMapper objectMapper;
+
+  public SdaKafkaProducerConfiguration(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
   @Bean
   @SuppressWarnings("java:S1452")
   public KafkaTemplate<String, ?> kafkaTemplate(ProducerFactory<String, ?> producerFactory) {
+    ((DefaultKafkaProducerFactory<?, ?>) producerFactory)
+        .setValueSerializer(new JsonSerializer<>(objectMapper));
```

Comment:
I have a bad feeling if we substitute the value serializer by a brand new instance. What if it was not a `JsonSerializer` at all? What if it was configured in a specific way?

Have you compared the approach mentioned in the ticket to use a `DefaultKafkaProducerFactoryCustomizer`? Do you have arguments for this approach compared to the mentioned approach?

Reply:
I'm pretty sure you would need to create your own `kafkaTemplate` bean anyways if you want to configure something, so I would assume it's ok to configure this default kafkaTemplate with this JsonSerializer

Reply:
I think this is the line of code where I've seen someone setting the value serializer before: https://github.com/SDA-SE/sda-spring-boot-commons/blob/master/sda-commons-starter-kafka/src/main/resources/org/sdase/commons/spring/boot/kafka/producer.properties#L2

Will users of the library override the value?
On the other hand: As you mentioned, with Spring's flexibility you should still be able to define your own KafkaTemplate and use a different serializer.

Reply:
Instead of configuring Kafka to use a specific object mapper. why don't we configure the default object mapper correctly? this would have the benefit that object mapper throughout the application has the same configuration and we don't need to work with different mappers in different modules

https://www.baeldung.com/spring-boot-customize-jackson-objectmapper

Depending on the specific time format it might also be possible to just adjust properties e.g.
`spring.jackson.date-format=`
`spring.jackson.joda-date-time-format=`
`spring.jackson.serialization.write-dates-as-timestamps=false`

https://docs.spring.io/spring-boot/docs/1.4.x/reference/html/common-application-properties.html